kong/clustering/services/sync/rpc.lua

```diff
@@ -175,6 +175,57 @@ local function is_rpc_ready()
 end
 
 
+local function lmdb_upsert(db, t, delta, opts)
+  local delta_type = delta.type
+  local delta_entity = delta.entity
+
+  -- upsert the entity
+  -- delete if exists
+  local old_entity, err = db[delta_type]:select(delta_entity)
+  if err then
+    return nil, err
+  end
+
+  local res
+  if old_entity then
+    res, err = delete_entity_for_txn(t, delta_type, old_entity, opts)
+    if not res then
+      return nil, err
+    end
+  end
+
+  res, err = insert_entity_for_txn(t, delta_type, old_entity, opts)
+  if not res then
+    return nil, err
+  end
+
+  return { delta_type, old_entity and "update" or "create", delta_entity, old_entity, }
+end
+
+
+local function lmdb_delete(db, t, delta, opts, is_full_sync)
+  local delta_type = delta.type
+
+  local old_entity, err = db[delta_type]:select(delta.pk, opts)
```

Comment:
I think that you changed the logic:
```
      local old_entity, err = db[delta_type]:select(delta_entity)
      if err then
        return nil, err
      end
```
why?

Reply:
fixed